share/hybrid/gpu-manager.c

```diff
@@ -2140,21 +2131,48 @@ static bool prime_set_discrete(int mode) {
 }
 
 
+static bool prime_set_rtd3(const struct device *device) {
+    _cleanup_fclose_ FILE *file = NULL;
+    _cleanup_fclose_ FILE *reset = NULL;
+    char reset_path[64];
+
+    /* TODO: Reset the device in kernel module */
+    sprintf(reset_path, "/sys/bus/pci/devices/%04x:%02x:%02x.%x/reset",
+            device->domain, device->bus, device->dev, device->func);
+    reset = fopen(reset_path, "w");
+    fprintf(reset, "1\n");
+    fflush(reset);
+
+    if (!is_module_loaded("bbswitch_dev")) {
+        load_module("bbswitch_dev");
+        while (access(bbswitch_dev_path, W_OK) != 0)
```

Comment:
Maybe check the status of load_module() here, as we don't want to wait forever for that file to be available, if loading the module fails.

Reply:
Yea, the sysfs should get properly generated before module loading returns, I'll update this.

Reply:
Force pushed a new one that addresses the issue.